### ISSUE TITLE
`hlint` doesn't recognize `CPP` language extension when `-XHaskell2010` is passed

### ISSUE THREAD

**Issue by RyanGlScott** [ndmitchell/hlint#1360]
If I run `hlint -XHaskell2010` on this module:

```hs
{-# LANGUAGE CPP #-}
{-# LANGUAGE MagicHash #-}
module Bug where

#if 1
import GHC.Exts

add# :: Int# -> Int# -> Int#
add# = (+#)
#endif
```

It will fail:

```
$ hlint -XHaskell2010 Bug.hs
Bug.hs:5:2: Error: Parse error: lexical error in pragma at character 'i'
Found:
    module Bug where
    
  > #if 1
    import GHC.Exts
  

1 hint
```

In order to work around this, I have to explicitly pass `-XCPP` to `hlint`:

```
$ hlint -XHaskell2010 -XCPP Bug.hs
No hints
```

This step feels unnecessary, however, because `Bug.hs` _already_ enables `CPP` at the top. Moreover, I didn't have to pass `-XMagicHash` to `hlint`, even though that is also required for the code to parse, so there appears to be a discrepancy in which flags one must explicitly pass to `hlint`.

**Comment by ndmitchell** [comment]
Thanks for the report, the PR from @shayne-fletcher in #1361 fixes it (and I think we can iterate a bit from there to get the right solution).
